include/drivers/clock_control/nrf_clock_control.h

```diff
@@ -94,4 +102,22 @@ int z_nrf_clock_calibration_count(void);
  */
 int z_nrf_clock_calibration_skips_count(void);
 
+/** @brief Get onoff service for given clock subsystem.
+ *
+ * @param sys Subsystem.
+ *
+ * @return Service handler or NULL.
+ */
+struct onoff_manager *z_nrf_clock_control_get_onoff(clock_control_subsys_t sys);
```

Comment:
@tbursztyka @mnkp @jdascenzio @jfischer-phytec-iot @galak @mbolivar-nordic feedback welcome on this new pattern: obtaining an instance to the on/off service through the existing driver API.

Reply:
I did not see any changes on the public API side. Instead, it exposes private features to this driver. I don't have much opinion on this then. If that's solving an important nordic use case but has no grounds for being generalized, the there is no other way. Fine by me.

Reply:
I agree. At dev-review there was no feedback that others had similar requirements, so it seems appropriate to do it this way, at least to get started.

Reply:
Yes.  The original design concept did not expect these structures to be exposed, but in this case it's the most sensible approach.  The expected alternative (provide custom request/release API specific to the subsystem) can be demonstrated where there's value added by the wrappers.

Reply:
#22409 had the proposal to make it more generic.